.github/workflows/integration-tdx.yml

```diff
@@ -0,0 +1,62 @@
+on:
+  push:
+    paths-ignore:
+      - "**.md"
+  pull_request:
+    paths-ignore:
+      - "**.md"
+  workflow_dispatch:
+
+name: Ingetration Test on TDX Server
+
+env:
+  AS: nasm
+  AR_x86_64_unknown_uefi: llvm-ar
+  CC_x86_64_unknown_uefi: clang
+  RUST_TOOLCHAIN: nightly-2022-04-07
+  TOOLCHAIN_PROFILE: minimal
+
+jobs:
+  system_compile:
+    name: Run TDX Integration Test
+    runs-on: [self-hosted, tdx]
+    timeout-minutes: 30
+
+    steps:
+      # Install first since it's needed to build NASM
+      - name: Install LLVM and Clang
+        uses: KyleMayes/install-llvm-action@v1
+        with:
+          version: "10.0"
+          directory: ${{ runner.temp }}/llvm
+
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: ${{ env.TOOLCHAIN_PROFILE }}
+          toolchain: ${{ env.RUST_TOOLCHAIN }}
+          override: true
+          components: rust-src, llvm-tools-preview
+
+      - name: Run cargo install cargo-xbuild
+        uses: actions-rs/cargo@v1
+        with:
+          command: install
+          args: cargo-xbuild
+
+      - name: Build TdShim
+        uses: actions-rs/cargo@v1
+        with:
+          command: xbuild
+          args: -p td-shim --target x86_64-unknown-uefi --release --features=main,tdx
+
+      - name: Build PE format payload
+        run: |
+          cargo xbuild -p td-payload --target x86_64-unknown-uefi --release --features=main,tdx
+          cargo run -p td-shim-tools --features=td-shim/default --bin td-shim-ld -- target/x86_64-unknown-uefi/release/ResetVector.bin target/x86_64-unknown-uefi/release/td-shim.efi target/x86_64-unknown-uefi/release/td-payload.efi -o target/x86_64-unknown-uefi/release/final-pe.bin
+
+      - name: Run Tests
+        run: bash sh_script/integration_tdx.sh -f target/x86_64-unknown-uefi/release/final-pe.bin
```

Comment:
Then it'd require changing the binary name here to `final-pe-boot-kernel.bin` instead.